### PR TITLE
feat: added kafka client and tiler client skeleton

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: src/serviceClients/clientsBase/kafkaClient.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      this.kafkaConfig.brokers = this.kafkaConfig.brokers.split(' ');
+    }
+
+    logger.log(
+      'info',
+      `Kafka manager created clientId=${this.kafkaConfig.clientId}, topic=${this.kafkaConfig.topic} brokers=${JSON.stringify(
+        this.kafkaConfig.brokers
+      )}`
+    );
+    const internalKafkaConfig: KafkaConfig = {
```
Comment (asafMasa) [inline src/serviceClients/clientsBase/kafkaClient.ts:27]: open issue about the ssl (same as in trigger - you should link it)

File: src/serviceClients/clientsBase/kafkaClient.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      brokers: this.kafkaConfig.brokers,
+    };
+    const kafka = new Kafka(internalKafkaConfig);
+    this.producer = kafka.producer({
+      createPartitioner: Partitioners.DefaultPartitioner,
+    });
+  }
+
+  protected async sendMessage(message: string): Promise<void> {
+    this.logger.log('debug', `sendMessage to kafka: message=${message}`);
```
Comment (asafMasa) [inline src/serviceClients/clientsBase/kafkaClient.ts:38]: why not logger.debug(?

File: src/serviceClients/clientsBase/kafkaClient.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    try {
+      await this.internalSendMessage(message);
+    } catch (error) {
+      this.logger.log('error', `Failed sending message to kafka, message=${message}`);
```
Comment (asafMasa) [inline src/serviceClients/clientsBase/kafkaClient.ts:42]: why not logger.error(?

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/serviceClients/clientsBase/kafkaClient.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+export abstract class KafkaClient {
+  protected producer: Producer;
+
+  public constructor(protected readonly logger: ILogger, protected readonly kafkaConfig: IKafkaConfig) {
+    if (typeof this.kafkaConfig.brokers === 'string') {
+      this.kafkaConfig.brokers = this.kafkaConfig.brokers.split(' ');
+    }
+
+    logger.log(
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply src/serviceClients/clientsBase/kafkaClient.ts:21]: there is no logger info in the new template

File: src/serviceClients/clientsBase/kafkaClient.ts (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply src/serviceClients/clientsBase/kafkaClient.ts:38]: there is no logger debug in the new template

File: src/serviceClients/clientsBase/kafkaClient.ts (hunk 3 above)
Comment (cl-shaharshavit-rnd) [reply src/serviceClients/clientsBase/kafkaClient.ts:42]: there is no logger error in the new template

**Review by asafMasa — APPROVED** [review]

**Pull request merged**